python/cuml/prims/stats/__init__.py

```diff
@@ -14,4 +14,4 @@
 # limitations under the License.
 #
 
-from cuml.prims.stats.covariance import cov
+from cuml.prims.stats.covariance import cov, cov_sparse
```

Comment:
I would really like to keep it this way. This function is just a bug fix for lager sp matrices. Whenever this bug gets resolved it would be very easy to kick `cov_sparse` and use `cov` again. It's also very specific for the needs of PCA.

Reply:
In my experience these kind of "temporary" work-arounds tend to stick around much longer than expected. The issue with adding a separate function is that a) any other code that might use the `cov` function now or in the future under these conditions will not automatically benefit from the fix and b) we add a function with overlapping and duplicated purpose.

IMO, my suggested change would it make even easier to remove the bug fix since the change would be completely transparent to the consuming function. So, I'm not sure that I follow your argument there. Happy to hear your thoughts!

In either case, we should clearly document this as a work-around and link to the corresponding issue to trigger a future removal.

Reply:
I 100% percent agree with the workaround flagging. I'm sorry if I was not clear enough. `cov_sparse` is not a fully functioned replacement of `cov`, eg. the mean is calculated and it only supports the same X. I you want me to code a fully functional cov_sparse replacement I can do this. Just give me a day or two. My main focus was in fixing sparse PCA.

Reply:
I'm sorry, I think there was a misunderstanding. What I am suggesting when I say "delegate" is to mark the `cov_sparse` function as private, i.e., rename it to `_cov_sparse()` and then simply call it within the `cov()` function under the appropriate conditions. In this way we avoid the increase in API surface. There is absolutely no need to replicate the full functionality of `cov()`, but  please let me know if I am missing something.

Reply:
That would be perfectly fine with me. I think I was the one that missed the point. Making them private makes perfect sense to me.

Reply:
Ok I had to expand `_cov_sparse` a bit because I don't know if and how the same case as in `PCA` comes up. In my limited testing it seems to work well.